### PR TITLE
Feature: adding optional sslmode parameter to postgres connection

### PR DESCRIPTION
**Pull request by dholleran-lendico**
resolves [#2152](https://github.com/fishtown-analytics/dbt/issues/2152)

### Description

added functionality to add 'sslmode' parameter to a postgres profile connection

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by dholleran-lendico** [comment]
hey @drewbanin , I'm not sure why the CLA check is failing, as I signed it, any ideas?

**Comment by beckjake** [comment]
@cla-bot check

**Comment by beckjake** [comment]
Hey @dholleran-lendico, thanks for your contribution! The CLA is a bit of a manual process on our end :)

Could you please resolve the merge conflict and while you're at it, include a link to this PR in the changelog?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 ### Features
 - Add a "docs" field to models, with a "show" subfield ([#1671](https://github.com/fishtown-analytics/dbt/issues/1671), [#2107](https://github.com/fishtown-analytics/dbt/pull/2107))
 - Add a dbt-{dbt_version} user agent field to the bigquery connector ([#2121](https://github.com/fishtown-analytics/dbt/issues/2121), [#2146](https://github.com/fishtown-analytics/dbt/pull/2146))
-
+- Adding optional "sslmode" parameter for postgres ([#2152](https://github.com/fishtown-analytics/dbt/issues/2152))
```
Comment (beckjake) [inline CHANGELOG.md]:
```suggestion
- Adding optional "sslmode" parameter for postgres ([#2152](https://github.com/fishtown-analytics/dbt/issues/2152), [#2154](https://github.com/fishtown-analytics/dbt/pull/2154))
```

File: plugins/postgres/dbt/adapters/postgres/connections.py [hunk 2]
```diff
@@ -89,6 +90,9 @@ def open(cls, connection):
             kwargs['options'] = '-c search_path={}'.format(
                 search_path.replace(' ', '\\ '))
 
+        if credentials.sslmode:
+             kwargs['sslmode'] = credentials.sslmode
```
Comment (beckjake) [inline plugins/postgres/dbt/adapters/postgres/connections.py]:
```suggestion
            kwargs['sslmode'] = credentials.sslmode
```

**Pull request merged**